<launch>
    <arg name="offline"          default="false"/>
    <include unless="$(arg offline)" 
        file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="align_depth" value="true"/>
        <arg name="linear_accel_cov" value="1.0"/>
        <arg name="unite_imu_method" value="linear_interpolation"/>
    </include>
    
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="ImuFilter">
        <param name="use_mag" type="bool" value="false" />
        <param name="_publish_tf" type="bool" value="false" />
        <param name="_world_frame" type="string" value="enu" />
        <remap from="/imu/data_raw" to="/camera/imu"/>
    </node>

    <include file="$(find rtabmap_ros)/launch/rtabmap.launch">
        <arg name="args" value="--delete_db_on_start"/>
        <arg name="rgb_topic" value="/camera/color/image_raw"/>
        <arg name="depth_topic" value="/camera/aligned_depth_to_color/image_raw"/>
        <arg name="camera_info_topic" value="/camera/color/camera_info"/>
        <arg name="depth_camera_info_topic" value="/camera/depth/camera_info"/>
        <arg name="rtabmapviz" value="false"/>
        <arg name="rviz" value="false"/>
    </include>

    <include file="$(find robot_localization)/launch/ukf_template.launch"/>
    <param name="/ukf_se/frequency" value="300"/>
    <param name="/ukf_se/base_link_frame" value="camera_link"/>
    <param name="/ukf_se/odom0" value="rtabmap/odom"/>
    <rosparam param="/ukf_se/odom0_config">[true,true,true,
                                            true,true,true,
                                            true,true,true,
                                            true,true,true,
                                            true,true,true]
    </rosparam>
    <param name="/ukf_se/odom0_relative" value="true"/>
    <param name="/ukf_se/odom0_pose_rejection_threshold" value="10000000"/>
    <param name="/ukf_se/odom0_twist_rejection_threshold" value="10000000"/>

    <param name="/ukf_se/imu0" value="/imu/data"/>
    <rosparam param="/ukf_se/imu0_config">[false, false, false,
                                           true,  true,  true,
                                           true,  true,  true,
                                           true,  true,  true,
                                           true,  true,  true]
    </rosparam>
    <param name="/ukf_se/imu0_differential" value="true"/>
    <param name="/ukf_se/imu0_relative" value="false"/>
    <param name="/ukf_se/use_control" value="false"/>
    <!-- <param name="/ukf_se/odom0_config" value="{true,true,true,}"/> -->

    <node name="object_detection" pkg="ros_openvino" type="object_detection" output="screen">
        <remap from="/object_detection/input_image" to="/camera/color/image_raw" />
        <remap from="/object_detection/input_depth" to="/camera/aligned_depth_to_color/image_raw" />
        <remap from="/object_detection/camera_info" to="/camera/aligned_depth_to_color/camera_info" />
        <param name="threshold"         value="0.5" />
        <param name="target"            value="MYRIAD" />
        <param name="model_network"     value="$(find ros_openvino)/models/FP16/mobilenet-ssd/mobilenet-ssd.xml" />
        <param name="model_weights"     value="$(find ros_openvino)/models/FP16/mobilenet-ssd/mobilenet-ssd.bin" />
        <param name="model_labels"      value="$(find ros_openvino)/models/FP16/mobilenet-ssd/mobilenet-ssd.labels" />
        <param name="model_colors"      value="$(find ros_openvino)/models/FP16/mobilenet-ssd/mobilenet-ssd.colors" />
        <param name="output_as_image"   value="true" />
        <param name="output_as_list"    value="true" />
        <param name="depth_analysis"    value="true" />
        <param name="output_as_markers" value="true" />
        <param name="output_as_makerslabel" value="true" />
        <param name="output_as_boxlist" value="false" />
    </node>

    <!-- <node name="semantic_tagging" pkg="ros_openvino" type="semantic_tagging" output="screen">
        <param name="labels"      value="$(find ros_openvino)/config/labels.txt" />
        <param name="colors"      value="$(find ros_openvino)/config/colors.txt" />
    </node>

    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find ros_openvino)/rviz/semantic_tagging.rviz" /> -->

</launch>